hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/domain/TokenTypeEnum.java

```diff
@@ -0,0 +1,42 @@
+package com.hedera.mirror.importer.domain;
+
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import java.util.Arrays;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@Getter
+@RequiredArgsConstructor
+public enum TokenTypeEnum {
+
+    FUNGIBLE_COMMON(0),
+    NON_FUNGIBLE_UNIQUE(1);
+
+    private final int id;
+
+    public static TokenTypeEnum fromId(int id) {
+        return Arrays.stream(TokenTypeEnum.values())
+                .filter(tokenTypeEnum -> tokenTypeEnum.id == id)
+                .findFirst()
+                .orElse(FUNGIBLE_COMMON);
+    }
```

Comment:
Since this is used in a hotspot, let's optimize by switching to a static map:
```suggestion
    private static final Map<Integer, TokenTypeEnum> ID_MAP = Arrays.stream(values()).collect(Collectors
            .toUnmodifiableMap(EntityTypeEnum::getId, Function.identity()));

    public static TokenTypeEnum fromId(int id) {
        return ID_MAP.getOrDefault(id, FUNGIBLE_COMMON);
    }
```

Reply:
Good idea, did this here and to TokenSupplyTypeEnum.